File: Examples/test-suite/python/template_templated_constructors_runme.py
from template_templated_constructors import *

t1 = TConstructor1(123)
t2a = TConstructor2()
t2b = TConstructor2(123)

tc1 = TClass1Int(123.4)
tc2a = TClass2Int()
tc2b = TClass2Int(123.4)

double_pair = DoublePair(1.1, 2.2)
short_pair = ShortPair(0, 1)
string_pair = StringPair("10", "11")
ip1 = IntPair()
ip2 = IntPair(20, 21)
ip3 = IntPair(ip1)
ip4 = IntPair(short_pair)
# These next two create an IntPair from factory function calls in Python, unlike Java which calls the IntPair constructor
ip5 = Pair(double_pair)
ip6 = MakeStringPair(string_pair)

File: Examples/test-suite/lua/template_templated_constructors_runme.lua
local v=require("template_templated_constructors")

local t1 = v.TConstructor1(123)
local t2a = v.TConstructor2()
local t2b = v.TConstructor2(123)
local tc1 = v.TClass1Int(123.4)
local tc2a = v.TClass2Int()
local tc2b = v.TClass2Int(123.4)
local double_pair = v.DoublePair(1.1, 2.2)
local short_pair = v.ShortPair(0, 1)
local string_pair = v.StringPair("10", "11")
--[[ TODO %template(IntPair) Pair<int, int> fail
v.IntPair()
v.IntPair(20, 21)
v.IntPair(ip1)
v.IntPair(short_pair)
]]
-- v.Pair(double_pair) TODO fails
-- v.MakeStringPair(string_pair) TODO %template(MakeStringPair) fails
